From 97c3295402d6654913c3821a662ff15593527b27 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:49:34 -0400
Subject: [ethernet] bnx2x: include irq.h for irqreturn_t definitions

Message-id: <1409946615-5862-30-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91572
O-Subject: [RHEL7.1 PATCH 29/70] net: bnx2x: include irq.h for irqreturn_t definitions
Bugzilla: 1089402
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402

Upstream commit:
commit df1efc2d3037334f4a868aca9ae329d67058bbfd
Author: Josh Boyer <jwboyer@fedoraproject.org>
Date:   Wed Apr 2 11:21:58 2014 -0400

    net: bnx2x: include irq.h for irqreturn_t definitions

    The bnx2x driver fails to build on ARM with:

    In file included from drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c:28:0:
    drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.h:243:1: error: unknown type name 'irqreturn_t'
     irqreturn_t bnx2x_msix_sp_int(int irq, void *dev_instance);
     ^
    drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.h:251:1: error: unknown type name 'irqreturn_t'
     irqreturn_t bnx2x_interrupt(int irq, void *dev_instance);
     ^

    Nothing in bnx2x_link.c or bnx2x_cmn.h is explicitly including the irq
    definitions, so we add an include of linux/irq.h to pick them up.

    Signed-off-by: Josh Boyer <jwboyer@fedoraproject.org>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.h b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.h
index beca592..0a46f93 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.h
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.h
@@ -21,6 +21,7 @@
 #include <linux/pci.h>
 #include <linux/netdevice.h>
 #include <linux/etherdevice.h>
+#include <linux/irq.h>
 
 #include "bnx2x.h"
 #include "bnx2x_sriov.h"
-- 
1.7.1